desugarer.go

```diff
@@ -182,13 +196,18 @@ func buildAnd(left ast.Node, right ast.Node) ast.Node {
 	return &ast.Binary{Op: ast.BopAnd, Left: left, Right: right}
 }
 
-func desugarForSpec(inside ast.Node, forSpec *ast.ForSpec) (ast.Node, error) {
+// inside is assumed to be already desugared (and cannot be desugared again)
+func desugarForSpec(inside ast.Node, forSpec *ast.ForSpec, objLevel int) (ast.Node, error) {
 	var body ast.Node
 	if len(forSpec.Conditions) > 0 {
 		cond := forSpec.Conditions[0].Expr
 		for i := 1; i < len(forSpec.Conditions); i++ {
 			cond = buildAnd(cond, forSpec.Conditions[i].Expr)
 		}
+		err := desugar(&cond, objLevel)
```

Comment:
Did cond and forSpec.Expr get forgotten about before?

Reply:
No, it was just desugared in two steps. First the forSpec was desugared then the result was desugared again, now without the comprehensions, which was clearly not ideal (both for performance and our sanity).